guides/testing.md

```diff
@@ -0,0 +1,163 @@
+# Testing Crystal Code
+
+Crystal comes with a full-feature spec library in the [`Spec` module](https://crystal-lang.org/api/latest/Spec.html). It provides the structure for writing executable examples of how your code should behave.
+
+Inspired by [Rspec](http://rspec.info/) it includes a domain specific language (DSL) that allows to write examples in a way similar to plain english..
+
+A basic spec looks something like this:
+
+```crystal
+require "spec"
+
+describe Array do
+  describe "#size" do
+    it "correctly reports the number of elements in the Array" do
+      [1, 2, 3].size.should eq 3
+    end
+  end
+
+  describe "#empty?" do
+    it "is empty when no elements are in the array" do
+      ([] of Int32).empty?.should be_true
+    end
+
+    it "is not empty if there are elements in the array" do
+      [1].empty?.should be_false
+    end
+  end
+end
+```
+
+## Anatomy of a spec file
+
+To use the spec module and DSL, you need to add `require "spec"` to your spec files. Many projects use a custom [spec helper](#spec-helper) which organizes these includes.
+
+In test files, specs are structured in example groups enclosed in `describe` and `context` sections. Typically a top level `describe` defines the outer unit (such as a class) to be tested by the spec. Further `describe` sections can be nested within the outer unit to specify smaller units under test (such as individual methods).
```

Comment:
Do you mean "specs are structured by example groups which are defined by `describe` and `context`? I'm not sure entirely how to interpret this paragraph. The core problem is that you haven't defined what an "example group" is.

Reply:
Yes, I'll take your rephrased version. I don't know if example groups need to be explained more detailed as *a group of examples*.

Reply:
I'd rather not mention "example groups" at all. I'd introduce `it` before `describe`/`context` to get the term "example" into the reader's head associated with `it` and then say:

> In test files, examples are structured in groups defined by `describe` and `context` blocks. [...]

or something similar.

Reply:
Sounds good. I think I had actually intended to put the `it` description first but somehow missed to reorder this.